_developer-content/import-api/guides/sequencing-data-for-the-import-api.md

```diff
@@ -0,0 +1,204 @@
+             "data":{
+                "id":1,
+                "name":"Finn",
+                "age":15,
+                "has_magic":false
+             }
+          }
+          ```
+
+          In this example, `sequence` is a Unix epoch in milliseconds that translates to `Tuesday, November 26, 2019 8:29:59 PM GMT-05:00`.
+
+      - title: "Record sequencing, Primary Keys, and loading"
+        anchor: "record-sequencing-primary-keys"
+        content: |
+          TODO - Content about Primary Keys and de-duping.
+
+  - title: "Defining record sequences"
+    anchor: "define-record-sequences"
+    summary: "How to define a record's sequence"
+    content: |
+      {% for subsection in section.subsections %}
+      - [{{ subsection.title }}](#{{ subsection.anchor }})
+      {% endfor %}
+
+    subsections:
+      - title: "Generating sequence values"
+        anchor: "generate-sequence-values"
+        content: |
+          When developing your Import API script, you'll need to define how values for the `sequence` property are generated. 
+
+          A simple solution is just to use the current timestamp, but before doing so, consider the following:
+
+          - **How frequently are the records being updated?** Records that are updated every few milliseconds can result in failure if records with identical key values are pushed simultaneously. This means that records with the same Primary Key values cannot be sent during the same clock resolution.
+
+             For example: If the resolution is measured in milliseconds, records with identical Primary Key values cannot be sent during the same millisecond.
+
+          - **Are records being pushed from multiple sources?** If so, the time clocks of these sources must be synchronized. This is especially important if different sources are pushing records to the same table.
+
+      - title: "Tracking a table's maximum sequence"
+        anchor: "track-table-maximum-sequence"
+        content: |
+          In addition to generating sequence values for records, you'll also need to track the maximum sequence value for each table being pushed to the Import API. Sequence values need to continually increase or records in the destination won't be updated with new data.
+
+          **Note**: A table's maximum sequence value must be tracked over time, and not for each session or request.
+
+  - title: "Examples"
+    anchor: "examples"
+    summary: "Some example scenarios"
+    content: |
+      {% include note.html type="single-line" content="**Note**: All example requests are using the Batch endpoint." %}
+
+    subsections:
+      - title: "TODO"
+        anchor: "todo"
+        content: |
+          This request contains three data points: One for record `id: 1`, and two for `id: 2`.
+
+          If these data points were received in this order, only data points 1 and 2 would continue to Stitch. Data point 3 would not, as its `sequence` value is less than the `sequence` value for data point 2.
```

Comment:
Hm... is this true? I don't see any code in the gate that checks that the sequence numbers are strictly increasing across the `messages` array, though maybe it's at load time where this would happen, not sure.

Reply:
@dmosorast yeah, that happens in the loader. Note that data point 3 would be omitted because its sequence number is less than the sequence number previously observed for another record _with the same primary key_.